### PR TITLE
fix(create-rslib): remove unnecessary React import

### PR DESCRIPTION
**Pull request by fi3ework**
## Summary

- Use `import type` for `React` when we're already using `automatic` jsx runtime
- Use `exclude` in `tsconfig.json` to make the test files are also included and get correct type resolve.

## Checklist

- [x] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: packages/create-rslib/fragments/base/react-ts/tsconfig.json [hunk 1]
```diff
@@ -10,5 +10,5 @@
     "moduleResolution": "bundler",
     "useDefineForClassFields": true
   },
-  "include": ["src"]
+  "exclude": ["node_modules"]
```
Comment (Timeless0911) [inline packages/create-rslib/fragments/base/react-ts/tsconfig.json:13]: Will this change affect the dts results? The best practice should be that the outer `tsconfig.json` is responsible for all type verification, and there is a `tsconfig.build.json` responsible for emitting dts files.

Comment (fi3ework) [reply packages/create-rslib/fragments/base/react-ts/tsconfig.json:13]: Agreed, and, without using project reference?

Comment (fi3ework) [reply packages/create-rslib/fragments/base/react-ts/tsconfig.json:13]: You're right, dts messed up, a lot of config file emitted. 🥲

Comment (Timeless0911) [reply packages/create-rslib/fragments/base/react-ts/tsconfig.json:13]: `the test files are also included and get correct type resolve`  I think this should configure in user side. Our tpl can ignore this.

File: packages/create-rslib/template-[react]-[storybook,vitest]-ts/tests/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "extends": "../tsconfig.json",
+  "include": [".", "../vitest.setup.ts"]
```
Comment (Timeless0911) [inline packages/create-rslib/template-[react]-[storybook,vitest]-ts/tests/tsconfig.json:3]: the storybook tpl can also add a separated `tsconfig.json`?

**Review by Timeless0911 — APPROVED** [review]

**Pull request merged**